app/views/devise/shared/_links.html.erb

```diff
@@ -0,0 +1,25 @@
+<%- if controller_name != 'sessions' %>
+  <%= link_to "Log in", new_session_path(resource_name) %><br />
+<% end -%>
+
+<%- if devise_mapping.registerable? && controller_name != 'registrations' %>
+  <%= link_to "Sign up", new_registration_path(resource_name) %><br />
+<% end -%>
+
+<%- if devise_mapping.recoverable? && controller_name != 'passwords' && controller_name != 'registrations' %>
```

Comment:
This has two problems, one is that the view should not be responsible for handling this much logic (Single Responsibility Principle). It's not the controller's responsibility, it might actually be a seperate library (you can build objects in Rails that aren't models, views or controllers after all!

The other problem is because this isn't encapsulated inside of a method that explains what it's doing, when an outsider looks at it, we have no idea what's going on